File: kt/problems/ReverseArray.kt
package problems

fun reverseArray(arr: Array<Int>): Array<Int> {
    val reversedArr = Array(arr.size) { 0 }
    var j = 0
    for (i in arr.size - 1 downTo 0) {
        reversedArr[j] = arr[i]
        j++
    }
    return reversedArr
}

fun main() {
    val arr = arrayOf(1, 2, 3, 4, 5)
    val reversedArr = reverseArray(arr)
    println(reversedArr.contentToString())
}

File: go/problems/reverseArray/reverseArray.go
package main

import "fmt"

func reverseArray(arr []int32) []int32 {
	var i int32
	var j int32 = int32(len(arr) - 1)

	for i < j {
		arr[i], arr[j] = arr[j], arr[i]
		i++
		j--
	}

	return arr
}

func main() {
	arr := []int32{1, 2, 3, 4, 5}

	fmt.Println(reverseArray(arr))
}
